docs/AAD-OAuth.md

```diff
@@ -373,3 +390,36 @@ export catalog=$(curl -s -H "Authorization: Bearer $acr_access_token" https://$r
 echo "Catalog"
 echo $catalog
 ```
+### Fetch helm index.yaml 
+
+```bash
+#!/bin/bash
+
+export registry=" --- you have to fill this out --- "
+export user=" --- you have to fill this out --- "
+export password=" --- you have to fill this out --- "
+
+export operation="/helm/v1/repo/index.yaml"
+ 
+export challenge=$(curl -vs https://$registry$operation 2>&1 | grep "Www-Authenticate:")
+echo "Challenge"
+echo $challenge
+ 
+export scope=$(echo $challenge | egrep -o 'scope=\"([^\"]*)\"' | egrep -o '\"([^\"]*)\"' | sed -e 's/^"//' -e 's/"$//')
+echo "Scope"
+echo $scope
+ 
+export credentials=$(echo -n "$user:$password" | base64 )
+ 
+export acr_access_token=$(curl -s -H "Content-Type: application/x-www-form-urlencoded" \
+ -H "Authorization: Basic $credentials" "https://$registry/oauth2/token?service=$registry&scope=$scope" | jq '.access_token' | sed -e 's/^"//' -e 's/"$//')
+echo "ACR Access Token"
+echo $acr_access_token
+ 
+#Retrieve the location header and strip the trailing \r for curl
+export URL=$(curl -sD - -H "Authorization: Bearer $acr_access_token" https://$registry$operation  | grep -Fi Location | awk '{print $2}' | tr -d '\r')
```

Comment:
If `Basic` auth is used, there is a simpler solution:

```
curl -sL --user "00000000-0000-0000-0000-000000000000:$acr_access_token" https://$registry$operation
```
or
```
curl -sL --user "$user:$password" https://$registry$operation
```

Basically, `curl` attaches the original headers to the redirect request, but authentication headers specified by `--user` are NOT attached to the redirect request, unless `--location-trusted` is also specified.